bench/CMakeLists.txt

```diff
@@ -3,17 +3,32 @@ set(BENCHMARK_ENABLE_TESTING OFF CACHE BOOL "Suppressing benchmark's tests" FORC
 add_subdirectory(benchmark)
 
 # remember to add src's here
-set(benchSrcs
-    operation_tree_bench.cpp
-    main_bench.cpp
+set(BenchSources
+    real_construction_bench.cpp
+    real_op_evaluation_bench.cpp
     )
+
+# Run main_bench to run all benchmarks
 # to run a subset of benchmarks, use:
-# ./main_bench --benchmark_filter=<regex>  
-add_executable(main_bench ${benchSrcs})
-target_include_directories(main_bench 
-    PRIVATE ./include
-    )
-target_link_libraries(main_bench 
-    PUBLIC Boost.Real
-    PUBLIC benchmark
-    )
+# ./main_bench --benchmark_filter=<regex>
+
+foreach(benchSrc ${BenchSources})
+    get_filename_component(benchName ${benchSrc} NAME_WE)
+    add_executable(${benchName} main_bench.cpp ${benchSrc})
+    target_include_directories(${benchName}
+        PRIVATE ./include
+        )
+     target_link_libraries(${benchName}
+         PUBLIC Boost.Real
+         PUBLIC benchmark
+         )
+endforeach(benchSrc)
+
+add_executable(main_bench main_bench.cpp ${BenchSources})
+    target_include_directories(main_bench
+        PRIVATE ./include
+        )
+     target_link_libraries(main_bench
+         PUBLIC Boost.Real
+         PUBLIC benchmark
+         )
```

Comment:
You may want to add a target to run all benchs and create reports in the future. 
Not needed now.